spec/game_looper_spec.rb

```diff
@@ -21,71 +21,128 @@ def initialize(marker)
 end
 
 class TestBoard
-  def mark_space(_message, _error_message); end
-end
+  attr_reader :values
+  attr_accessor :move_history
 
-class TestDisplay end
+  def initialize(values: nil)
+    @values = values
+    @move_history = []
+  end
 
-class TestPrompt
-  def call(_message, _error_message); end
+  def mark_space(token, space)
+    move_history << "Board with #{token} in space #{space}"
+  end
 end
 
 class TestOutcomeCheckerForGameLooper
-  def initialize(board:)
+  attr_accessor :board, :win_values, :draw_values
+
+  def initialize(board:, win_values:, draw_values:)
     @board = board
+    @win_values = win_values
+    @draw_values = draw_values
   end
-end
 
-describe 'Game Looper' do
-  context '#loop' do
-    it 'loops until there is a draw' do
-      console = TestConsole.new
-      prompt = TestPrompt.new
-      board = TestBoard.new
-      display = TestDisplay.new
-      outcome_checker = TestOutcomeCheckerForGameLooper.new(board:)
-      player_one = TestPlayer.new('X')
-      player_two = TestPlayer.new('O')
-      players = [player_one, player_two]
+  def win?
+    win_values.shift
+  end
 
-      allow(outcome_checker).to receive(:draw?).and_return(false, false, false, false, true)
-      allow(outcome_checker).to receive(:win?).and_return(false, false, false, false, false)
+  def draw?
+    draw_values.shift
+  end
+end
 
-      allow(display).to receive(:present).and_return('board', 'board', 'board_with_draw')
+class TestUI
+  attr_reader :output, :input
 
-      game_looper = GameLooper.new(console:, prompt:, board:, display:,
-                                   players:, outcome_checker:)
+  def initialize(input:)
+    @output = []
+    @input = input
+  end
 
-      game_looper.loop
+  def display_board(_board)
+    output << 'board displayed'
+  end
 
-      expected_console_messages = %w[board board board_with_draw]
+  def prompt
+    input.shift
+  end
+end
 
-      expect(console.messages).to eq(expected_console_messages)
+describe 'Game Looper' do
+  describe '#loop' do
+    context 'until there is a draw' do
+      it 'loops' do
+        test_input = []
+        test_win_values = []
+        test_draw_values = []
+        ui = TestUI.new(input: test_input)
+        board = TestBoard.new
+        outcome_checker = TestOutcomeCheckerForGameLooper.new(board:, win_values: test_win_values,
+                                                              draw_values: test_draw_values)
+        player_one = TestPlayer.new('X')
+        player_two = TestPlayer.new('O')
+        players = [player_one, player_two]
+
+        allow(outcome_checker).to receive(:draw?).and_return(false, false, true)
```

Comment:
I might be misunderstanding, but were these bool values meant to be in the the `test_win_values` and `test_draw_values`?